httpcore5/src/test/java/org/apache/hc/core5/net/TestURIBuilder.java

```diff
@@ -813,4 +813,32 @@ public void testNormalizeSyntax() throws Exception {
                 new URIBuilder("http:///../../.././").normalizeSyntax().build().toASCIIString());
     }
 
+    @Test
+    public void testIpv6Host() throws Exception {
+        final URI uri = new URIBuilder("https://[::1]:432/path").build();
+        Assert.assertEquals(432, uri.getPort());
+        Assert.assertEquals("https", uri.getScheme());
+        Assert.assertEquals("[::1]", uri.getHost());
+        Assert.assertEquals("/path", uri.getPath());
+    }
+
+    @Test
+    public void testIpv6HostWithPortUpdate() throws Exception {
+        // Updating the port clears URIBuilder.encodedSchemeSpecificPart
+        // and bypasses the fast/simple path which preserves input.
+        final URI uri = new URIBuilder("https://[::1]:432/path").setPort(123).build();
+        Assert.assertEquals(123, uri.getPort());
+        Assert.assertEquals("https", uri.getScheme());
+        Assert.assertEquals("[::1]", uri.getHost());
```

Comment:
See my comment above.

Reply:
There's nothing we can do in this case because we're operating on the constructed `URI` object, not the `URIBuilder`. `URIBuilder.getHost` does return `::1`.

Reply:
I've also updated the tests to check URIBuilder getters in addition to URI getters -- this should make the differences between URIBuilder.getHost and URI.getHost more obvious.